pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/UnpublishVersion.java

```diff
@@ -121,32 +128,41 @@ protected void doRun(SlingHttpServletRequest request, PostResponse response, Lis
             if(canUnPublish) {
                 serviceResourceResolver = serviceResourceResolverProvider.getServiceResourceResolver();
             }
-            Module module = serviceResourceResolver.getResource(request.getResource().getPath()).adaptTo(Module.class);
+            Optional<? extends Document> document;
+            if(serviceResourceResolver.getResource(request.getResource().getPath()).adaptTo(Document.class).getResourceType().equalsIgnoreCase("pantheon/assembly")){
+                Assembly assembly = serviceResourceResolver.getResource(request.getResource().getPath()).adaptTo(Assembly.class);
+                document = Optional.ofNullable(assembly);
+            }
+            else{
+                Module module = serviceResourceResolver.getResource(request.getResource().getPath()).adaptTo(Module.class);
+                document = Optional.ofNullable(module);
+
+            }
```

Comment:
Again, unnecessary, you should just be able to `.adaptTo(Document.class)` and the code should still function.